reflections.go

```diff
@@ -177,6 +197,25 @@ func fields(obj interface{}, deep bool) ([]string, error) {
 	return allFields, nil
 }
 
+// Converts a map datatype to struct datatype
+// obj must be a pointer to a structure
+func MapToStruct(obj map[string]interface{}, result interface{}) error {
```

Comment:
I can see how this could prove useful in certain scenarios, but I don't think this functionality quite match the scope of this library. Please drop this from this PR. I'm happy to discuss the functionality in a separate issue if you think I'm wrong, but I won't merge this PR with this function present 🙇🏻